### PR TITLE
Add definitions for react-transition-group

### PR DESCRIPTION
**Pull request by LKay**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/react-transition-group/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Type definitions for react-transition-group 1.1
+// Project: https://github.com/reactjs/react-transition-group
+// Definitions by: Asana <https://asana.com>,
```
Comment (andy-ms) [inline types/react-transition-group/index.d.ts]: Do you need to add all these authors?

Comment (LKay) [inline types/react-transition-group/index.d.ts]: I based this on already existing typings for legacy projects (`react-addons-transition-group` and `react-addons-css-transition-group`) and didn't want to be an asshole who removes credits. :)

Comment (andy-ms) [inline types/react-transition-group/index.d.ts]: But you're the one who wrote the definitions for *this* package, right?

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**